src/main/java/it/gov/pagopa/pu/debtpositions/service/statusalign/StatusRulesHandler.java

```diff
@@ -0,0 +1,87 @@
+package it.gov.pagopa.pu.debtpositions.service.statusalign;
+
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public abstract class StatusRulesHandler<E extends Enum<E>> {
+
+  private final E syncStatus;
+  private final E paidStatus;
+  private final E unpaidStatus;
+  private final E expiredStatus;
+  private final E cancelledStatus;
+  private final E reportedStatus;
+  private final E invalidStatus;
+
+  protected StatusRulesHandler(E syncStatus, E paidStatus, E unpaidStatus, E expiredStatus, E cancelledStatus, E reportedStatus, E invalidStatus) {
+    this.syncStatus = syncStatus;
+    this.paidStatus = paidStatus;
+    this.unpaidStatus = unpaidStatus;
+    this.expiredStatus = expiredStatus;
+    this.cancelledStatus = cancelledStatus;
+    this.reportedStatus = reportedStatus;
+    this.invalidStatus = invalidStatus;
+  }
+
+  public <T, S, C extends Enum<C>> void updateEntityStatus(T entity,
+                                                           Function<T, List<S>> childStatusExtractor,
+                                                           Function<List<S>, C> statusDeterminer,
+                                                           BiConsumer<T, C> statusUpdater,
+                                                           BiConsumer<T, C> repositoryUpdater) {
+
+    List<S> childStatuses = childStatusExtractor.apply(entity);
+    C newStatus = statusDeterminer.apply(childStatuses);
+    statusUpdater.accept(entity, newStatus);
+    repositoryUpdater.accept(entity, newStatus);
+  }
```

Comment:
instead of request this lambda, why don't you declare abstract methods?
The generic T, S and C could be generic for the entire class
```suggestion
  public void updateEntityStatus(T entity) {

    List<S> childStatuses = getChildrenStatuses(entity);
    C newStatus = calculateNewStatus(childStatuses);
    setStatus(entity, newStatus);
    storeStatus(entity, newStatus);
  }
  
  abstract List<S> getChildrenStatuses(T entity);
  ...
```